tests/functional/adapter/test_grants.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from dbt.tests.adapter.grants.test_model_grants import BaseModelGrants
+from dbt.tests.adapter.grants.test_incremental_grants import BaseIncrementalGrants
+from dbt.tests.adapter.grants.test_invalid_grants import BaseInvalidGrants
+from dbt.tests.adapter.grants.test_seed_grants import BaseSeedGrants
+from dbt.tests.adapter.grants.test_snapshot_grants import BaseSnapshotGrants, snapshot_schema_yml
+
+
+my_snapshot_sql = """
+{% snapshot my_snapshot %}
+    {{ config(
+        check_cols='all', unique_key='id', strategy='check',
+        target_database=database, target_schema=schema
+    ) }}
+    select 1 as id, cast('blue' as {{ type_string() }}) as color
+{% endsnapshot %}
+""".strip()
```

Comment:
Without this override, I was getting this error upon the 2nd of 2 consecutive snapshots:
```
failed to find conversion function from "unknown" to text
```

### Root cause
Old versions of Postgres (and apparently current versions of Redshift) have a known issue with detecting types of untyped constants [[1]](https://github.com/OHDSI/SqlRender/issues/66)[[2]](https://stackoverflow.com/questions/18073901/failed-to-find-conversion-function-from-unknown-to-text)[[3]](https://www.postgresql.org/message-id/65937bea0901060104n399cdec8ye5e2b8e247e5139a@mail.gmail.com)

Reply:
Optionally, we could add this (slightly) more verbose version to dbt-core.

On that note, is there syntax for type-casting that we could enable for core + adapters?

I'd prefer less verbose syntax like this:
```
select 1 as id, {{ cast_string('blue') }} as color
```
over this:
```
select 1 as id, cast('blue' as {{ type_string() }}) as color
```

The former would also allow some adapters to render more pithy SQL like:
```
'blue'::text
```
rather than:
```
cast('blue' as text)
```

Reply:
@dbeatty10 I think we could add a cross-db `{{ cast() }}` macro to `dbt-core`, which returns `column::type` where it's supported, or `cast(column as type)` otherwise / by default.

For now, I did the simpler version in https://github.com/dbt-labs/dbt-core/pull/5452/commits/7f499ba2c6d71545bfab9023ff3d182d9b100c98

Reply:
Awesome, I'll try the relevant update and see if it works.